txselector/txselector.go

```diff
@@ -574,18 +577,37 @@ func (txsel *TxSelector) processL2Txs(tp *txprocessor.TxProcessor,
 // l1CoordinatorTxs array, and then the PoolL2Tx is added into the validTxs
 // array.
 func (txsel *TxSelector) processTxToEthAddrBJJ(validTxs []common.PoolL2Tx,
-	selectionConfig txprocessor.Config, nL1UserTxs int, l1CoordinatorTxs []common.L1Tx,
-	positionL1 int, l2Tx common.PoolL2Tx) (*common.PoolL2Tx, *common.L1Tx,
-	*common.AccountCreationAuth, error) {
+	selectionConfig txprocessor.Config, nL1UserTxs int, l1UserFrozenTxs,
+	l1CoordinatorTxs []common.L1Tx, positionL1 int, l2Tx common.PoolL2Tx) (
+	*common.PoolL2Tx, *common.L1Tx, *common.AccountCreationAuth, error) {
 	// if L2Tx needs a new L1CoordinatorTx of CreateAccount type, and a
 	// previous L2Tx in the current process already created a
 	// L1CoordinatorTx of this type, in the DB there still seem that needs
 	// to create a new L1CoordinatorTx, but as is already created, the tx
 	// is valid
-	if checkAlreadyPendingToCreate(l1CoordinatorTxs, l2Tx.TokenID, l2Tx.ToEthAddr, l2Tx.ToBJJ) {
+	if checkPendingToCreateL1CoordTx(l1CoordinatorTxs, l2Tx.TokenID, l2Tx.ToEthAddr, l2Tx.ToBJJ) {
 		return &l2Tx, nil, nil, nil
 	}
 
+	// check if L2Tx receiver account will be created by a L1UserFrozenTxs
+	// (in the next batch, the current frozen queue). In that case, the L2Tx
+	// will be discarded at the current batch, even if there is an
+	// AccountCreationAuth for the account, as there is a L1UserTx in the
+	// frozen queue that will create the receiver Account.  The L2Tx is
+	// discarded to avoid the Coordinator creating a new L1CoordinatorTx to
+	// create the receiver account, which will be also created in the next
+	// batch from the L1UserFrozenTx, ending with the user having 2
+	// different accounts for the same TokenID. The double account creation
+	// is supported by the Hermez zkRollup specification, but it was decided
+	// to mitigate it at the TxSelector level for the explained cases.
+	if checkPendingToCreateFrozenQueue(l1UserFrozenTxs, l2Tx.TokenID, l2Tx.ToEthAddr, l2Tx.ToBJJ) {
+		return nil, nil, nil, fmt.Errorf("L2Tx discarded at the current batch," +
```

Comment:
Bear in mind that this error message will appear in the Info field.  I think it's very long!

Reply:
Shortened the error message. It was so long initially as this is a behavior that is not the 'default by protocol' behavior, but is a request to mitigate some particular cases, so the idea was to give as much details as possible on why the L2Tx was being discarded (even that technically with the protocol spec in the hand should be possible to be forged).
Now the `Info` message is shortened, and the details continue to be in the comment of the code, so the debugger will be able to find why the L2Tx is being discarded in those cases.